src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -197,6 +205,9 @@ async Task EnsureLong()
                 }
                 catch (Exception e)
                 {
+                    if (cancellationToken.IsCancellationRequested)
```

Comment:
This can move into the switch below, with a case for OperationCanceledException when cancellationToken.IsCancellationRequested. Probably a good idea to add a super short comment that this is user cancellation.